dataframely/_serialization.py

```diff
@@ -118,3 +121,367 @@ def object_hook(self, dct: dict[str, Any]) -> Any:
+    ) -> None:
+        """This method stores frames as individual parquet files.
+
+        Args:
+            df: DataFrame containing the data to be stored.
+            kwargs: The "file" kwarg is required to specify where data is stored.
+                It should point to a parquet file. If hive partitioning is used,
+                it should point to a directory.
+                The "metadata" kwarg is supported to pass a dictionary of parquet
+                metadata.
+                Additional keyword arguments are passed to polars.
+        """
+        file = kwargs.pop("file")
+        metadata = kwargs.pop("metadata", {})
+        df.write_parquet(
+            file,
+            metadata={**metadata, SCHEMA_METADATA_KEY: serialized_schema},
+            **kwargs,
+        )
+
+    def scan_frame(self, **kwargs: Any) -> tuple[pl.LazyFrame, SerializedSchema | None]:
+        """Lazily read single frames from parquet.
+
+        Args:
+            kwargs: The "source" kwarg is required to specify where data is stored.
+                Other kwargs are passed to polars.
+        """
+        source = kwargs.pop("source")
+        lf = pl.scan_parquet(source, **kwargs)
+        metadata = _read_serialized_schema(source)
+        return lf, metadata
+
+    def read_frame(self, **kwargs: Any) -> tuple[pl.DataFrame, SerializedSchema | None]:
+        """Eagerly read single frames from parquet.
+
+        Args:
+            kwargs: The "source" kwarg is required to specify where data is stored.
+                Other kwargs are passed to polars.
+        """
+        source = kwargs.pop("source")
+        df = pl.read_parquet(source, **kwargs)
+        metadata = _read_serialized_schema(source)
+        return df, metadata
+
+    # ------------------------------ Collections ---------------------------------------
+    def sink_collection(
+        self,
+        dfs: dict[str, pl.LazyFrame],
+        serialized_collection: SerializedCollection,
+        serialized_schemas: dict[str, str],
+        **kwargs: Any,
+    ) -> None:
+        """Stream multiple frames to parquet.
+
+        Args:
+            dfs: See base class.
+            serialized_collection: See base class.
+            serialized_schemas: See base class.
+            kwargs: The "directory" kwarg is required to specify where data is stored.
+        """
+        path = Path(kwargs.pop("directory"))
+
+        # The collection schema is serialized as part of the member parquet metadata
+        kwargs["metadata"] = kwargs.get("metadata", {}) | {
+            COLLECTION_METADATA_KEY: serialized_collection
+        }
+
+        for key, lf in dfs.items():
+            destination = (
+                path / key if "partition_by" in kwargs else path / f"{key}.parquet"
+            )
+            self.sink_frame(
+                lf,
+                serialized_schema=serialized_schemas[key],
+                file=destination,
+                **kwargs,
+            )
```

Comment:
Potentially for a follow-up PR: `sink_parquet` has a `lazy=True` parameter (see https://docs.pola.rs/api/python/stable/reference/api/polars.LazyFrame.sink_parquet.html). This allows us to create a separate lazy frame for writing all collection members, then call `collect_all` for maximum speedup ;)